### PR TITLE
RR-428 - Add missing prisonId fields

### PR DESCRIPTION
**Pull request by mattwills-moj-digital**
This PR adds the `createdAtPrison` and `lastUpdatedAtPrison` fields.

### CODE REVIEW

File: domain/induction/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/induction/PreviousQualifications.kt [hunk 1]
```diff
@@ -10,14 +11,17 @@ import java.time.Instant
  * been asked about their education).
  */
 data class PreviousQualifications(
-  val educationLevel: HighestEducationLevel,
+  val reference: UUID,
+  val educationLevel: HighestEducationLevel?,
```
Comment (mattwills-moj-digital) [inline domain/induction/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/induction/PreviousQualifications.kt:15]: This was touched on in the last meeting - it's not mandatory.

File: domain/induction/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/induction/FutureWorkInterests.kt [hunk 2]
```diff
@@ -11,13 +12,16 @@ import java.time.Instant
...
-  val createdBy: String?,
-  val createdByDisplayName: String?,
-  val createdAt: Instant?,
-  val lastUpdatedBy: String?,
-  val lastUpdatedByDisplayName: String?,
-  val lastUpdatedAt: Instant?,
+  val createdBy: String,
+  val createdByDisplayName: String,
+  val createdAt: Instant,
+  val createdAtPrison: String,
```
[the remark this answers is not in the document]
Comment (nathanrussell-moj-digital) [reply domain/induction/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/induction/FutureWorkInterests.kt:20]: No, I think your cardinality for the "audit" fields at the domain is wrong.
Have a look at our other domain classes (like `Goal`), but these "JPA" fields (I know they are not jpa at this level) are all optional
The reason they are optional is that when you create a new domain object (eg. a Goal), we create the domain object first before sending it into the JpaPersistenceAdapter which is the thing that actually saves it to the database and where JPA does its magic.
So, it is not until the domain object has been sent into the JpaPersistenceAdapter, saved, returned as a JPA entity, then mapped back into a domain object, that we have these audit fields.

[the remark this answers is not in the document]
Comment (mattwills-moj-digital) [reply domain/induction/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/induction/FutureWorkInterests.kt:20]: Ah good point, I was only thinking about the response 🤦‍♂️

[the remark this answers is not in the document]
Comment (mattwills-moj-digital) [reply domain/induction/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/induction/FutureWorkInterests.kt:20]: Fixed 👍

**Review by nathanrussell-moj-digital — APPROVED** [review]
LGTM 👍

**Pull request merged**